src/main/java/seedu/address/model/course/Course.java

```diff
@@ -0,0 +1,86 @@
+package seedu.address.model.course;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Course in the address book.
+ * Holds details about a course, such as its code, which follows a specific format.
+ * The course code is expected to adhere to the format defined by {@link #VALIDATION_REGEX},
+ * which ensures it matches the typical NUS course code format. This format includes two
+ * initial uppercase letters, followed by four digits, and an optional trailing uppercase letter.
+ * Note: The course code is not marked as final to allow for the possibility of changes to
+ * the course code after the course object has been created. This flexibility accommodates
+ * scenarios where course codes might need to be updated or corrected.
+ */
+public class Course {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Course code should follow the format \"XX1234Y\", Y is optional";
```

Comment:
We can consider using a constant for the format instead of a magic literal here:
```suggestion
    public static final String COURSE_CODE_FORMAT = "XX1234Y";
    public static final String MESSAGE_CONSTRAINTS =
            "Course code should follow the format \"" + COURSE_CODE_FORMAT + "\", Y is optional";
```